nova/virt/vmwareapi/vm_util.py

```diff
@@ -1757,9 +1772,15 @@ def get_stats_from_cluster_per_host(session, cluster):
                      "hardware.cpuPkg", "hardware.cpuInfo",
                      "config.featureCapability",
                     ])
+    hosts = {}
     with vutil.WithRetrieval(session.vim, result) as objects:
-        return dict(_process_host_stats(obj, host_reservations_map)
-                    for obj in objects if hasattr(obj, "propSet"))
+        for host_obj in objects:
+            if hasattr(host_obj, "propSet"):
+                props = propset_dict(host_obj.propSet)
+                props["stats"] = _process_host_stats(host_obj,
+                                                     host_reservations_map)
```

Comment:
Why do we keep the properties but also save them processed under `stats`? Do we need them in raw later on? It would be helpful if you could add a comment explaining that.

Is it just for keeping the `name` property? We could add that in `_process_host_stats`. We could also add the `moref_value` as additional key there. Then we could even keep the old code of this function ;)

Reply:
We do add `hypervisor_hostname` to it though in `host.py`'s `_merge_stats()` and afaics you're only interested in adding the name, so you could add `hypervisor_hostname` from the beginning and only overwrite it for the cluster-wide-aggregated data.